### DISCUSSION TITLE
Guide for onboarding developers?

### DISCUSSION THREAD

**Discussion by swingtempo** [immich-app/immich#14944] [Q&A]
Is there a guide for onboarding developers? I saw a bug in the web client and would like to take a crack at fixing it. I'm not too worried about the Svelte/Typescript part. I'm using Windows, FWIW. I'm able to build and run the `/web` code. Ideally, I would like to just point my SPA to a production server that I'm running. That's the part I can't figure out.  I tried calling:
```
  setBaseUrl("http://**myserver**:2283/api");
  setApiKey("**myApiKey**");
```
and ran Chrome with noCORS. However I'm not able to get past the login page.

It may seem a little complicated - I would love to know the workflows of devs that are currently developing on the project so I can help contribute.

**Comment by alextran1502** [comment]
Hello, you can start from here. 

https://immich.app/docs/developer/architecture

FWiW, developing on Linux ensures the best experience. 

We create a simple make command to run the development by typing `make dev-update` from the root of the project to start everything, including hot reload

**Comment by swingtempo** [comment]
Thanks!

I didn't see that set of web pages.

________________________________
